### PR TITLE
For #23526 - Add telemetry for Contile interactions

### LINKED ISSUE #23526: Contile Telemetry
https://mozilla-hub.atlassian.net/browse/FNXV2-18724

### PR DESCRIPTION
**Pull request by MozillaNoah**
Closes #23526 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by MozillaNoah** [comment]
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
- How often are users that see the contile clicking on it?
- How often are users opening a contile (directly, in new tab, in new private tab)?
- What's the ratio of users dismissing a contile to clicking on it?
- What percent of DAU have contile turned off?
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
  * This will help us understand usage patterns around the Contile feature
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives. 
4. Can current instrumentation answer these questions?
  * No
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. 
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
User clicks on a Contile | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/23526
User long presses a Contile | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/23526
User opens a Contile link in a private tab via the long press context menu | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/23526
User opens Settings via the long press context menu  | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/23526
User opens “Sponsors & your privacy” via the long press context menu  | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/23526
User disables/enables the Contile feature | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/23526

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/fenix 
7. How long will this data be collected? Choose one of the following:
  * One year (ending on February 1st, 2023), with the option to renew at that point. 
8. What populations will you measure?
  * All channels, all locales, all countries 
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism. 
10. Please provide a general description of how you will analyze this data.
  * Glean and Amplitude. 
11. Where do you intend to share the results of your analysis?
  *  Only on Glean, Amplitude, and with mobile teams. 
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

File: app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt [hunk 1]
```diff
@@ -945,6 +974,15 @@ class DefaultSessionControlControllerTest {
         }
     }
 
+    @Test
+    fun `WHEN handleTopSiteSettingsClicked is called THEN the interaction is reported`() {
+        createController().handleTopSiteSettingsClicked()
+
+        verify {
+            metrics.track(Event.TopSiteContileSettings)
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt:982]: Can we group this with the existing test?

File: app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt [hunk 2]
```diff
@@ -958,6 +996,63 @@ class DefaultSessionControlControllerTest {
         }
     }
 
+    @Test
+    fun `WHEN handleSponsorPrivacyClicked is called THEN the interaction is reported`() {
+        createController().handleSponsorPrivacyClicked()
+
+        verify {
+            metrics.track(Event.TopSiteContilePrivacy)
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt:1004]: Same as above, let's just group this with the existing test

**Comment by gabrielluong** [comment]
We typically use the present for the commit message (s/Added/Add)

**Comment by gabrielluong** [comment]
Wondering if we also had any telemetry requirement with the turning on/off the feature in the Homepage settings.

**Comment by MozillaNoah** [comment]
> Wondering if we also had any telemetry requirement with the turning on/off the feature in the Homepage settings.

it's actually already being reported since the original implementation story via the preference change listener `CustomizeHomeMetricsUpdater`

**Comment by rocketsroger** [comment]
> # Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
>     * How often are users that see the contile clicking on it?
> 
>     * How often are users opening a contile (directly, new private tab)?
> 
>     * What's the ratio of users dismissing a contile to clicking on it?
> 
>     * What percent of DAU have contile turned off?
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
> 
> 
>     * This will help us understand usage patterns around the Contile feature
> 
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * There are no other alternatives.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No
> 
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.    
>        Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> User clicks on a Contile 	Category 2 - interaction data 	#23526
> User long presses a Contile 	Category 2 - interaction data 	#23526
> User opens a Contile link in a private tab via the long press context menu 	Category 2 - interaction data 	#23526
> User opens Settings via the long press context menu 	Category 2 - interaction data 	#23526
> User opens “Sponsors & your privacy” via the long press context menu 	Category 2 - interaction data 	#23526
> User disables/enables the Contile feature 	Category 2 - interaction data 	#23526
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
>     * [https://dictionary.telemetry.mozilla.org/apps/fenix   ](https://dictionary.telemetry.mozilla.org/apps/fenix%E2%80%A8)
> 
> 
>     7. How long will this data be collected? Choose one of the following:
> 
> 
>     * One year (ending on February 1st, 2023), with the option to renew at that point.
> 
> 
>     8. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries
> 
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean and Amplitude.
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean, Amplitude, and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tools.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection set to end or be renewed by 2023-02-01

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Review by rocketsroger — APPROVED** [review]

File: app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt [hunk 3]
```diff
@@ -958,6 +993,54 @@ class DefaultSessionControlControllerTest {
         }
     }
 
+    @Test
+    fun `WHEN handleOpenInPrivateTabClicked is called WITH a TopSite#Provided site THEN Event#TopSiteOpenContileInPrivateTab is reported`() {
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt:997]: WITH doesn't need to fully capitalized. It's just GIVEN, WHEN, THEN

File: app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt [hunk 4]
```diff
@@ -958,6 +993,54 @@ class DefaultSessionControlControllerTest {
...
+        )
+        createController().handleOpenInPrivateTabClicked(topSite)
+
+        verify {
+            metrics.track(Event.TopSiteOpenContileInPrivateTab)
+        }
+    }
+
+    @Test
+    fun `WHEN handleOpenInPrivateTabClicked is called WITH TopSite#Default or TopSite#Pinned or TopSite#Frecent site THEN Event#TopSiteOpenInPrivateTab is reported`() {
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt:1015]:
```suggestion
    fun `WHEN handleOpenInPrivateTabClicked is called with Default or Pinned or Frecent top site THEN TopSiteOpenInPrivateTab event is reported`() {
```

Perhaps a simplification to the #

**Pull request merged**